From d36be2f5a0a8011786e345ee0905c324690f1a43 Mon Sep 17 00:00:00 2001
From: Kuppuswamy Sathyanarayanan <sathyanarayanan.kuppuswamy@intel.com>
Date: Wed, 10 Sep 2014 13:35:34 -0700
Subject: input: ft5x06: Removed unnecessary delay in reset

Changed reset pulse delay from 6 ms to 1 ms as specified
in datasheet. Also reduced additional delays added after
reset and mode change functions.

These changes will reduce boot and resume latency.

Signed-off-by: Kuppuswamy Sathyanarayanan <sathyanarayanan.kuppuswamy@intel.com>
---
 drivers/input/touchscreen/ft5x06_ts.c | 2 +-
 include/linux/input/ft5x06_ts.h       | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/input/touchscreen/ft5x06_ts.c b/drivers/input/touchscreen/ft5x06_ts.c
index e6edd27..4f315ae 100644
--- a/drivers/input/touchscreen/ft5x06_ts.c
+++ b/drivers/input/touchscreen/ft5x06_ts.c
@@ -234,7 +234,7 @@ static int ft5x0x_wake_device(struct ft5x0x_ts_data *ts_data)
 {
 	/* wake the device by wake line*/
 	gpiod_set_value(ts_data->gpio_reset, 0);
-	usleep_range(6000, 7000);
+	usleep_range(1000, 1500);
 	gpiod_set_value(ts_data->gpio_reset, 1);
 	msleep(FT5X0X_RST_MS);
 
diff --git a/include/linux/input/ft5x06_ts.h b/include/linux/input/ft5x06_ts.h
index b77b473..61fe3c4 100644
--- a/include/linux/input/ft5x06_ts.h
+++ b/include/linux/input/ft5x06_ts.h
@@ -5,8 +5,8 @@
 #define SCREEN_MAX_Y		1280
 #define PRESS_MAX		255
 
-#define FT5X0X_RST_MS		500
-#define FT5X0X_DEVMODE_MS	500
+#define FT5X0X_RST_MS		5
+#define FT5X0X_DEVMODE_MS	5
 
 #define CFG_MAX_TOUCH_POINTS	10
 #define FT_PRESS		8
-- 
1.9.1

